web/src/components/core/SelectorTable.jsx

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from "react";
+import { Table, Thead, Tr, Th, Tbody, Td, ExpandableRowContent } from "@patternfly/react-table";
+
+/**
+ * An object for sharing data across nested maps
+ *
+ * Since function arguments are always passed by value, an object passed by
+ * sharing is needed for sharing data that might be mutated from different
+ * places, as it is the case of the rowIndex prop here.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions#passing_arguments
+ *
+ * @typedef {object} SharedData
+ * @property {number} rowIndex - The current row index, to be incremented each time a table row is generated.
+ */
+
+/**
+ * @typedef {object} SelectorTableColumn
+ * @property {string} name - The column header text
+ * @property {(object) => React.Element} value - A function receiving
+ *   the item to work with and returning the column value.
```

Comment:
Thinking it twice, maybe using `header` over `name` and `content` over `value` makes that _column object_ easier to understand.

```js
{
  header: "The column header / label",
  content: (item) => "The column content"
}
```

vs

```js
{
  name: "The column header / label",
  value: (item) => "The column content"
}
```